### PR TITLE
Network link address API doc fixes

### PR DESCRIPTION
**Pull request by jukkar**

### CODE REVIEW

**Review by nashif — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

File: doc/reference/networking/net_linkaddr.rst [hunk 1]
```diff
@@ -1,11 +1,14 @@
 .. _net_linkaddr_interface:
 
-Link Adreess
-############
+Link Address Handling
```
Comment (pfalcon) [inline doc/reference/networking/net_linkaddr.rst]: I hope this is not treated as nitpick. But these things are actually called "link layer address", e.g. we use "lladdr" in the source. I'd suggest to fix it at least here in the heading, if not in the rest of docs.

Comment (nashif) [reply doc/reference/networking/net_linkaddr.rst]: if you do not want it to be treated as nitpick, request a change please.

Comment (pfalcon) [reply doc/reference/networking/net_linkaddr.rst]: Well, it's up to @jukkar whether to address it now, or leave for later. We have a lot of doc tweaks to make anyway, can't do it all at once.

Comment (jukkar) [reply doc/reference/networking/net_linkaddr.rst]: Sure, I can modify the title. I will send a new version.

**Review by dbkinder — APPROVED** [review]

**Pull request merged**